client/src/keyboard-controls.tsx

```diff
@@ -1,56 +1,120 @@
 import makeRequest from "./utils/request/makeRequest";
 
-const keys = new Map();
+/*
+ * Controls for controlling the robot are as follows:
+ * 
+ * Drive:
+ *  ArrowUp: drive forward
+ *  ArrowDown: drive backward
+ *  ArrowLeft: turn counterclockwise
+ *  ArrowRight: turn clockwise
+ *  C (hold): reduce speed
+ *  V (hold): increase speed
+ * 
+ * Arm:
+ *  Q/A: arm base
+ *  W/S: shoulder
+ *  E/D: elbow
+ *  R/F: forearm
+ *  T/G: diff left
+ *  Y/H: diff right
+ *  U/J: hand
+ */
+
+const driveControlKeys = ["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight", "c", "v"];
+const armControlKeys = ["q", "a", "w", "s", "e", "d", "r", "f", "t", "g", "y", "h",
+    "u", "j"];
+const armMotors = ["arm_base", "shoulder", "elbow", "forearm", "diffleft",
+    "diffright", "hand"];
+
+const pressedKeys = new Map();
+// Storing the PWM values allows us to only necessary arm motor requests.
+const armPwmValues = new Array(7).fill(0.0);
 
 export function addKeyboardListeners() {
     document.addEventListener("keydown", onKeyPress);
     document.addEventListener("keyup", onKeyRelease);
 }
 
 function onKeyPress(event) {
-    const key = String.fromCharCode(event.keyCode);
-    if (!keys[key]) {
-      keys[key] = true;
-      updateDrive();
+    // Multiple events can be fired for just one key press if the user holds the
+    // key down, so we need to check if the key is already pressed.
+    if (!pressedKeys[event.key]) {
+        pressedKeys[event.key] = true;
+        if (driveControlKeys.includes(event.key)) {
+            // preventDefault() prevents users from scrolling through the page when
+            // moving the robot using arrow keys.
+            event.preventDefault();
```

Comment:
This wasn't working on my machine because after the first keypress event, this part of the code doesn't run. I fixed it and pushed to this branch.